blocks/query-panel/index.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import CategorySelect from './category-select';
+import RangeControl from '../inspector-controls/range-control';
+import SelectControl from '../inspector-controls/select-control';
+
+const DEFAULT_MIN_ITEMS = 1;
+const DEFAULT_MAX_ITEMS = 100;
+
+export default function QueryPanel( {
+	category,
+	numberOfItems,
+	order,
+	orderBy,
+	maxItems = DEFAULT_MAX_ITEMS,
+	minItems = DEFAULT_MIN_ITEMS,
+	onCategoryChange,
+	onNumberOfItemsChange,
+	onOrderChange,
+	onOrderByChange,
+} ) {
+	return [
+		( onOrderChange || onOrderByChange ) && (
+			<SelectControl
+				key="query-panel-select"
+				label={ __( 'Order by' ) }
+				value={ `${ orderBy }/${ order }` }
+				options={ [
+					{
+						label: __( 'Newest to Oldest' ),
+						value: 'date/desc',
+					},
+					{
+						label: __( 'Oldest to Newest' ),
+						value: 'date/asc',
+					},
+					{
+
+						/* translators: label for ordering posts by title in ascending order, translation may be required for languages using non latin alphabet */
```

Comment:
Kinda feel we can drop the rest of the comment after the first comma, that's really up to the translator to judge for themselves, and the first bit should be sufficient for understanding the intent of the string.

Reply:
Personal preference would be to drop the newline prior to this comment, feels in line with this ESLint rule we have enabled:

https://eslint.org/docs/rules/padded-blocks (never)